src/inmanta/module.py

```diff
@@ -2401,6 +2401,15 @@ def get_plugin_dir(self) -> Optional[str]:
         """
         raise NotImplementedError()
 
+    def _list_python_files(self, plugin_dir: str) -> Iterator[str]:
```

Comment:
What if both the pyc and py file are present? 

I think this can happen by accident and should be tested

Reply:
pyc has precedence over py and pyc is invalidated based on timestamp.

Normally pyc files are always created in a __pycache__, so this will only happen when created without our build command.

What else would you like to test?

Reply:
So suppose a module v1.0.0 is byte-compiled but v2.0.0 isn't (or the other way around), both the compiler and the agent code loader should load the latest version as expected?

Reply:
By where is this filtering done? 

if I look around: 
1. on the exporter code path, I think both the pyc and py files are sent to the server
2. on the loading path, this will cause duplicate loading in 'load_plugins' I think


I also think that if both the py and pyc file are sent over, but the mtime of the files are not also sent, the pyc file will always be recycled. (as per @arnaudsjs research in this matter, iirc, the mtime of the py file is inside the pyc file)

So I would indeed only use the pyc if there is no py file

Reply:
I'm not completely following. Suppose the yang module is used on an orchestrator somewhere right now. When this is merged we create a byte-compiled artifact for it, that also contains some new functionality. How do we make sure this new functionality will be available if we prefer the `.py` files? There are no `.py` files anymore for the new version, but the ones from the old version are not cleaned up.

Reply:
As it is implemented now, we will select a loader that loads either pyc or py depending on what the server returned. So that means that if v+1 switches to python it will use the python code.

Reply:
done + tests